lib/protocol_9p_server.ml

```diff
@@ -246,6 +246,7 @@ module Make(Log: Protocol_9p_s.LOG)(FLOW: V1_LWT.FLOW)(Filesystem: Protocol_9p_f
     LowLevel.expect_version ~write_lock reader writer
     >>*= fun (tag, v) ->
     let msize = min msize v.Request.Version.msize in
+    let msize = min msize Protocol_9p_buffered9PReader.max_message_size in
```

Comment:
Ah, indeed, I read this as removal and addition instead of just addition. Sorry about that.